src/marked.js

```diff
@@ -139,45 +139,111 @@ marked.defaults = defaults;
  */
 
 marked.use = function(extension) {
-  const opts = merge({}, extension);
-  if (extension.renderer) {
-    const renderer = marked.defaults.renderer || new Renderer();
-    for (const prop in extension.renderer) {
-      const prevRenderer = renderer[prop];
-      renderer[prop] = (...args) => {
-        let ret = extension.renderer[prop].apply(renderer, args);
-        if (ret === false) {
-          ret = prevRenderer.apply(renderer, args);
+  if (!Array.isArray(extension)) { // Wrap in array if not already to unify processing
+    extension = [extension];
+  }
+
+  const opts = merge({}, ...extension);
+  const extensions = marked.defaults.extensions || { renderers: {}, childTokens: {} };
+  let hasExtensions;
+
+  extension.forEach((pack) => {
+    // ==-- Parse "addon" extensions --== //
+    if (pack.extensions) {
+      hasExtensions = true;
+      pack.extensions.forEach((ext) => {
+        if (ext.renderer && ext.name) { // Renderers must have 'name' property
+          const prevRenderer = extensions.renderers?.[ext.name] || null;
```

Comment:
` || null` seems redundant

```suggestion
          const prevRenderer = extensions.renderers?.[ext.name];
```

Reply:
done.